engine/execution/computation/manager.go

```diff
@@ -120,6 +147,16 @@ func (e *Manager) ComputeBlock(
 		return nil, fmt.Errorf("failed to execute block: %w", err)
 	}
 
+	toInsert := programs
+
+	// if we have item from cache and there were no changes
+	// insert it under new block, to prevent long chains
+	if fromCache != nil && !programs.HasChanges() {
```

Comment:
Would taking a parent cache be useful?
```suggestion
	if fromCache != nil && !programs.HasChanges(fromCache) {
```

Reply:
No, `HasChanges` checks no only for items in the map, but also if the special `clear` flag has been set, so you can't really compare them